src/main.js

```diff
@@ -322,17 +322,29 @@ class MiniGraphCard extends LitElement {
     `;
   }
 
+  renderLoader() {
+    return html`
+        <div style="display: flex;">
+             <ha-circular-progress indeterminate style="margin: auto;"></ha-circular-progress>
+        </div>
+    `;
+  }
+
   renderGraph() {
+    const ready = this.entity[0] && this.Graph[0]._history !== undefined;
+
     return this.config.show.graph ? html`
       <div class="graph">
-        <div class="graph__container">
-          ${this.renderLabels()}
-          ${this.renderLabelsSecondary()}
-          <div class="graph__container__svg">
-            ${this.renderSvg()}
-          </div>
-        </div>
-        ${this.renderLegend()}
+        ${ready ? html`
+            <div class="graph__container">
+              ${this.renderLabels()}
+              ${this.renderLabelsSecondary()}
+              <div class="graph__container__svg">
+                ${ready ? this.renderSvg() : this.renderLoader()}
+              </div>
+            </div>
+            ${this.renderLegend()}
+        ` : this.renderLoader()}
```

Comment:
If the `renderLoader` became too trivial as per the other comments above, consider to literally write the ha-circular-progress directly here.

Reply:
Agreed

Reply:
Inlined